src/actionHandlers/commands/slowMode.ts

```diff
@@ -0,0 +1,38 @@
+import { ApplicationCommandOptionType, PermissionsBitField } from 'discord.js';
+
+import type { Command } from './types';
+
+export const slowMode: Command = (interaction) => {
+  const { options, channel, member } = interaction;
+
+  const duration = options.getNumber('duration');
+
+  // check if the member is an administrator
+  if (!(member.permissions as PermissionsBitField).has('Administrator')) return;
+
+  // remove slow-mode from channel
+  if (duration == 0) {
+    channel
+      .setRateLimitPerUser(duration)
+      .then(() => interaction.reply({ content: `Slow mode has been removed`, ephemeral: true }));
+    return;
+  }
+
+  // adding slow-mode to the channel
+  channel
+    .setRateLimitPerUser(duration, 'reason')
```

Comment:
Here too if u take a close look u mentioned this line two times. Kinda redundant yk. so I suggest moving it always set the duration then send the message later